src/manifest.rs

```diff
@@ -136,3 +144,21 @@ pub fn write_package_json(
 pub fn get_crate_name(path: &str) -> Result<String, Error> {
     Ok(read_cargo_toml(path)?.package.name)
 }
+
+fn has_cdylib(path: &str) -> bool {
+    match read_cargo_toml(path).unwrap().lib {
+        Some(lib) => match lib.crate_type {
```

Comment:
the nested pattern matching here feels unnecessarily verbose -- is there a more succinct way to express this?

Reply:
You probably want to use [and_then](https://doc.rust-lang.org/std/option/enum.Option.html#method.and_then) in order to do this as well as `map_or`. If you haven't looked at Option or Result's docs definitely do so as you can find all kinds of neat methods to help!

The code will probaby end up looking like this if you want to make it more succinct

```rust
// If nothing goes wrong when it reads then we have the value wrapped up in an Ok.
// If something goes wrong the ? will return the error early
Ok(read_cargo_toml(path)?.lib
    // If the value is some we can then doe something with the value
    .and_then(|lib| {
         // Use the map_or function for Options to map the inner value to a boolean. If the
         // value for lib.crate_type is None then default to false. Return the boolean as the
         // Option is consumed with map_or
         lib.crate_type.map_or(|types| types.iter().any(|s| s == "cdylib), false)
     }))
```
If that's not clear tell me why and I can explain it a bit better! :D

Reply:
Actually you'll want to use map_or instead of and_then, with a default value of false now that I'm looking at this again.